# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/mine/hotfix-maxpathlen.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- mine-0.23/libtar-1.2.11/compat/basename.c.vanilla	2025-12-27 19:52:25.000000000 +0100
+++ mine-0.23/libtar-1.2.11/compat/basename.c	2025-12-27 19:52:35.000000000 +0100
@@ -35,6 +35,10 @@
 #include <string.h>
 #include <sys/param.h>
 
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 4096 /* For GNU Hurd */
+#endif
+
 char *
 openbsd_basename(path)
 	const char *path;
--- mine-0.23/libtar-1.2.11/compat/dirname.c.vanilla	2025-12-27 19:52:51.000000000 +0100
+++ mine-0.23/libtar-1.2.11/compat/dirname.c	2025-12-27 19:53:02.000000000 +0100
@@ -35,6 +35,10 @@
 #include <string.h>
 #include <sys/param.h>
 
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 4096 /* For GNU Hurd */
+#endif
+
 char *
 openbsd_dirname(path)
 	const char *path;
--- mine-0.23/libtar-1.2.11/lib/libtar.h.vanilla	2025-12-27 19:51:46.000000000 +0100
+++ mine-0.23/libtar-1.2.11/lib/libtar.h	2025-12-27 19:52:08.000000000 +0100
@@ -24,6 +24,9 @@
 {
 #endif
 
+#ifndef MAXPATHLEN
+#define MAXPATHLEN 4096 /* For GNU Hurd */
+#endif
 
 /* useful constants */
 #define T_BLOCKSIZE		512
